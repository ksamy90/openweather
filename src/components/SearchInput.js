import React, { useState } from "react";
import { connect } from "react-redux";
import { startFetchWeather } from "../actions/weather";

export class SearchInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      city: "",
    };
  }
  inputChange = (e) => {
    const city = e.target.value;
    this.setState(() => ({ city }));
  };
  refreshPage = () => {
    window.location.reload(false);
  };
  onFormSubmit = (e) => {
    e.preventDefault();
    if (this.state.city === "") {
      return;
    } else {
      this.props.fetchWeather(this.state.city);
      this.setState(() => ({ city: "" }));
    }
  };

  render() {
    return (
      <div className="search-input">
        <button className="refreshbtn" onClick={this.refreshPage}>
          refresh
        </button>
        <form onSubmit={this.onFormSubmit}>
          <input
            type="text"
            placeholder="get weather forecast in your city"
            value={this.state.city}
            onChange={this.inputChange}
            className="input-field"
          />
          <button>search weather</button>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchWeather: (city) => dispatch(startFetchWeather(city)),
  };
};

export default connect(undefined, mapDispatchToProps)(SearchInput);
