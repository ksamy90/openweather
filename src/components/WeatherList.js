import React from "react";
import { connect } from "react-redux";
import { Sparklines, SparklinesLine } from "react-sparklines";

export const WeatherList = (props) => {
  console.log(props);
  const temperatures = props.weather.map((weather) => {
    return weather.list.map((data) => {
      return data.main.temp;
    });
  });
  return (
    <div className="weather-list">
      <h1>Temperature</h1>
      {props.weather.map((cityData) => {
        return (
          <div key={cityData.city.id}>
            <h2>{cityData.city.name}</h2>
            <div>
              <span className="tempdata">{temperatures[0][0]} kelvin</span>
              <span className="tempdata">{temperatures[0][1]} kelvin</span>
              <span className="tempdata">{temperatures[0][2]} kelvin</span>
            </div>
          </div>
        );
      })}
      {temperatures && (
        <Sparklines
          data={temperatures[0]}
          limit={5}
          height={120}
          width={180}
          margin={5}
        >
          <SparklinesLine color="blue" />
        </Sparklines>
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    weather: state.weather,
  };
};

export default connect(mapStateToProps)(WeatherList);
