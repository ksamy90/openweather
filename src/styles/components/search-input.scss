.search-input {
  margin-left: 400px;
  margin-top: 70px;
}

.input-field {
  height: 40px;
  width: 250px;
  border-radius: 10px;
  border: none;
}

button {
  margin-left: 10px;
  height: 45px;
  width: 200px;
  border-radius: 10px;
  border: none;
}

.refreshbtn {
  margin-bottom: 20px;
  margin-left: 160px;
}
