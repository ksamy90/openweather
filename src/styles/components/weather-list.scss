.weather-list {
  margin: 0 auto;
  height: 300px;
  width: 400px;
}

.tempdata {
  margin-right: 30px;
}
