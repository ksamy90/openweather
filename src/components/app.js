import React from "react";
import SearchInput from "./searchInput";
import WeatherList from "./WeatherList";

export default function App() {
  return (
    <div>
      <SearchInput />
      <WeatherList />
    </div>
  );
}
