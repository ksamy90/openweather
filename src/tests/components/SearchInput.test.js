import React from "react";
import { shallow } from "enzyme";
import { SearchInput } from "../../components/SearchInput";

test("should render SearchInput correctly", () => {
  const wrapper = shallow(<SearchInput />);
  expect(wrapper).toMatchSnapshot();
});

test("should set input state on input change", () => {
  const value = "Munich";
  const wrapper = shallow(<SearchInput />);
  wrapper.find("input").at(0).simulate("change", {
    target: { value },
  });
  expect(wrapper.state("city")).toBe(value);
});
