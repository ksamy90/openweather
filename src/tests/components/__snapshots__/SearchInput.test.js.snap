// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchInput correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="refreshbtn"
          onClick={[Function]}
>
          refresh
</button>,
        <form
          onSubmit={[Function]}
>
          <input
                    className="input-field"
                    onChange={[Function]}
                    placeholder="get weather forecast in your city"
                    type="text"
                    value=""
          />
          <button>
                    search weather
          </button>
</form>,
      ],
      "className": "search-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "refresh",
          "className": "refreshbtn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "refresh",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="input-field"
              onChange={[Function]}
              placeholder="get weather forecast in your city"
              type="text"
              value=""
/>,
            <button>
              search weather
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "onChange": [Function],
              "placeholder": "get weather forecast in your city",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "search weather",
            },
            "ref": null,
            "rendered": "search weather",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="refreshbtn"
            onClick={[Function]}
>
            refresh
</button>,
          <form
            onSubmit={[Function]}
>
            <input
                        className="input-field"
                        onChange={[Function]}
                        placeholder="get weather forecast in your city"
                        type="text"
                        value=""
            />
            <button>
                        search weather
            </button>
</form>,
        ],
        "className": "search-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "refresh",
            "className": "refreshbtn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "refresh",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="input-field"
                onChange={[Function]}
                placeholder="get weather forecast in your city"
                type="text"
                value=""
/>,
              <button>
                search weather
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "onChange": [Function],
                "placeholder": "get weather forecast in your city",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "search weather",
              },
              "ref": null,
              "rendered": "search weather",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
